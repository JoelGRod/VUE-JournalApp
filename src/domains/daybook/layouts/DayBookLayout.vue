<template>
    <h1>DayBook Layout</h1>
</template>