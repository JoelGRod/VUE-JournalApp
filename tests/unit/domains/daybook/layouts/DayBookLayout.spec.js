import { shallowMount } from "@vue/test-utils";

import DayBookLayout from "@/domains/daybook/layouts/DayBookLayout";
import createVuexStore from "../../../mock-data/mock-store";

describe("Testing: DayBookLayout.vue", () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const store = createVuexStore({
    user: {
      name: "Testing Store",
      email: "testing@store.com",
    },
    status: "authenticated",
    idToken: "abc",
    refreshToken: "xyz",
  });

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(DayBookLayout, {
      global: {
        plugins: [store],
        mocks: {
          $router: mockRouter,
        },
      },
    });
  });

  test("should match with the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  // test("onLogout should close session and redirect", async () => {
  //   // Act
  //   const spyLogout = await jest.spyOn(wrapper.vm, "logout");
  //   await wrapper.vm.onLogout();
  //   // Assert
  //   // expect(store.state.auth).toEqual({
  //   //   user: null,
  //   //   status: "not-authenticated",
  //   //   idToken: null,
  //   //   refreshToken: null,
  //   // });
  //   expect(spyLogout).toHaveBeenCalled();
  //   expect(mockRouter.push).toHaveBeenCalledWith({ name: "Login" });
  // });
});
