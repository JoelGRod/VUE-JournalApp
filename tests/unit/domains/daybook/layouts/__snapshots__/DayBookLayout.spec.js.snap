// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: DayBookLayout.vue should match with the snapshot 1`] = `
<navbar-component-stub username="Testing Store"></navbar-component-stub>
<div class="d-flex">
  <div class="col-4">
    <entry-list-component-stub></entry-list-component-stub>
  </div>
  <div class="col">
    <router-view></router-view>
  </div>
</div>
`;
